src/NexusMods.App.UI/Pages/CollectionDownload/CollectionComponents.cs

```diff
@@ -0,0 +1,213 @@
+using CollectionDownloadEntity = NexusMods.Abstractions.NexusModsLibrary.Models.CollectionDownload;
+
+public static class CollectionColumns
+{
+    [UsedImplicitly]
+    public sealed class Actions : ICompositeColumnDefinition<Actions>
+    {
+        public static int Compare<TKey>(CompositeItemModel<TKey> a, CompositeItemModel<TKey> b) where TKey : notnull
+        {
+            // TODO:
+            return 0;
+        }
+
+        public const string ColumnTemplateResourceKey = nameof(CollectionColumns) + "_" + nameof(Actions);
+        public static readonly ComponentKey NexusModsDownloadComponentKey = ComponentKey.From(ColumnTemplateResourceKey + "_" + nameof(CollectionColumns) + "_" + "NexusModsDownload");
+        public static readonly ComponentKey ExternalDownloadComponentKey = ComponentKey.From(ColumnTemplateResourceKey + "_" + nameof(CollectionColumns) + "_" + "ExternalDownload");
+        public static readonly ComponentKey ManualDownloadComponentKey = ComponentKey.From(ColumnTemplateResourceKey + "_" + nameof(CollectionColumns) + "_" + "ManualDownload");
+        public static readonly ComponentKey InstallComponentKey = ComponentKey.From(ColumnTemplateResourceKey + "_" + nameof(CollectionColumns) + "_" + "Install");
+
+        public static string GetColumnHeader() => "Action";
+        public static string GetColumnTemplateResourceKey() => ColumnTemplateResourceKey;
+    }
+}
+
+public static class CollectionComponents
+{
+    public abstract class ADownloadAction<TSelf, TEntity> : ReactiveR3Object, IItemModelComponent<TSelf>, IComparable<TSelf>
+        where TSelf : ADownloadAction<TSelf, TEntity>, IItemModelComponent<TSelf>, IComparable<TSelf>
+        where TEntity : struct, IReadOnlyModel<TEntity>
+    {
+        public int CompareTo(TSelf? other)
+        {
+            if (other is null) return 1;
+            return JobStatusComparer.Instance.Compare(DownloadStatus.Value, other.DownloadStatus.Value);
+        }
+
+        private readonly IDisposable _activationDisposable;
+        protected ADownloadAction(
+            TEntity downloadEntity,
+            Observable<JobStatus> downloadJobStatusObservable,
+            Observable<bool> isDownloadedObservable)
+        {
+            _activationDisposable = this.WhenActivated((downloadEntity, downloadJobStatusObservable, isDownloadedObservable), static (self, state, disposables) =>
+            {
+                var (_, downloadJobStatusObservable, isDownloadedObservable) = state;
+
+                downloadJobStatusObservable.CombineLatest(isDownloadedObservable, static (a, b) => (a, b)).Subscribe(self, static (tuple, self) =>
+                {
+                    var (downloadStatus, isDownloaded) = tuple;
+                    self._downloadStatus.Value = downloadStatus;
+                    self._buttonText.Value = GetButtonText(isDownloading: downloadStatus == JobStatus.Running, isDownloaded);
+                }).AddTo(disposables);
```

Comment:
If you downloaded all the items and open the page, this is never triggered. 
`CombineLatest` usually does that when one of the passed observables doesn't have starting value.
All passed observables need an initial value